vxlan/vxlan.go

```diff
@@ -0,0 +1,277 @@
+// Package vxlan provides the capabilities to create a VXLAN overlay network
+package vxlan
+
+import (
+	"net"
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/pkg/errors"
+	"github.com/rancher/go-rancher-metadata/metadata"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	changeCheckInterval = 5
+	metadataURL         = "http://rancher-metadata/2016-07-29"
+	vxlanMACRange       = "0E:00:00:00:00:00"
+	vxlanPort           = 4789
+)
+
+const (
+	DefaultBridgeName             = "docker0"
+	DefaultVxlanMTU               = 1500
```

Comment:
Can you please test on AWS with a x2large instance where the MTU of eth0 is 9000+ and then set the VTEP MTU to 9000?
Last time when I tried to change the MTU of vtep to 9000, I couldn't. Want to make sure we are able to change it using `netlink` library.